src/deterministic_secure_broadcast.rs

```diff
@@ -13,6 +13,23 @@ use serde::{Deserialize, Serialize};
 pub enum Error {
```

Comment:
related things:
1. Shouldn't this impl std::error::Error?  (which implies impl for std::fmt::Display)
2. would be nice to have human display messages for these error (and the validations)
3. perhaps consider use of crate `thiserror` to make 1 and 2 easy and clean.

Reply:
Bit the bullet and went to set up `thiserror` and it was quite nice! Thanks for pushing it.

I've updated the Error enums for both DSB and BFT-Membership to use `thiserror`

Reply:
looks great!